src/textfield/textfield-foundation.tsx

```diff
@@ -32,6 +33,11 @@ export const useTextFieldFoundation = (props: TextFieldProps) => {
   const setFloatingLabel = (api: FloatingLabelApi | null) =>
     (floatingLabel.current = api);
 
+  const { content: characterCountContent } =
+    useTextFieldCharacterCountFoundation({
+      apiRef: props.characterCount ? setCharacterCounter : undefined
+    });
+
```

Comment:
#777  This was incredibly annoying to diagnose and fix. The breakdown:
- The way RMWC is setup, the foundations are initialized when components render. This is fine 99% of the time.
- In this particular case, the TextfieldFoundation requires the TextfieldCharacterCount to be initialized. This creates a really annoying race condition because the character count component will be mounted after the text field.
- Solution, get rid of the current internal TextfieldCharacterCount component and instead just initialize the foundation inside of useTextfieldFoundation and just export the actual character count number
- Extra annoying, hooks cannot be initialized conditionally, so we pass in undefined for the api ref if character count isn't in play which effectively short circuits the character count.
- And it works 🤦‍♂️